src/shaders/shader.h

```diff
@@ -3,28 +3,35 @@
 #include <GL/glew.h>
 #include <glm/glm.hpp>
 
-#include "model/material.h"
-#include "model/geometry.h"
+#include "scene/scene_info.h"
+#include "model/mesh.h"
 
 class Shader {
-public:
+private:
     GLuint shader_id;
-    glm::mat4 V, P, mesh_model;
-    glm::vec3 cam_pos;
-
-    Shader(GLuint shader_id);
+public:
+    void set_id(GLuint id) { shader_id = id; }
```

Comment:
Thing is, we have the shaders as static instances. And their IDs aren't determined until GL is initialized.